src/clj/chronograph/handlers/google_auth.clj

```diff
@@ -26,40 +26,51 @@
            (.getPayload)
            (into {})))
 
-(defn- redirect-uri [{:keys [redirect-uri
-                             client-id
-                             response-type
-                             scope
-                             login-endpoint] :as oauth-config}]
+(defn- auth-code-uri [{:keys [redirect-uri
+                              client-id
+                              response-type
+                              scope
+                              login-endpoint]}]
   (format "%s?redirect_uri=%s&client_id=%s&scope=%s&response_type=%s"
           login-endpoint
           (codec/url-encode redirect-uri)
           (codec/url-encode client-id)
           (codec/url-encode scope)
           (codec/url-encode response-type)))
 
+(defn- redirect-uri [client-type]
+  (let [{:keys [web-redirect-uri desktop-redirect-uri]} (get-in config/config
+                                                                [:oauth :google])]
+    (if (= "desktop" client-type)
```

Comment:
Would it be better to define predicates for client-types? It seems like we would have a definite set of client-types.

Reply:
There are only two, `desktop` and `web`. We could extract a predicate regardless.

Reply:
Actually nvm...don't think it's worth extracting predicates since there is only one place where we are checking this.